File: src/my_project/easy_problems/from201to250/maximum_number_of_balloons.py
from typing import List 

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        
        # answer 
        answer = 0
        
        # dictionaty to add the total occurences of each word in text
        temp_dict = dict()
        
        len_text = len(text) 
        
        for word in "balon":
            temp_dict[word] = 0
        
        # calculate the many times each word appears in text
        for i in range(len_text):
            temp_dict[text[i]] = 0
        
        for i in range(len_text):
            temp_dict[text[i]] += 1     
            
        # the total amoung of occurrences of each letter to form the word balloon
        ballon_occurrences = []        
          
        ballon_occurrences.append(temp_dict['b'])
        ballon_occurrences.append(temp_dict['a'])
        ballon_occurrences.append(temp_dict['l']//2)
        ballon_occurrences.append(temp_dict['o']//2)
        ballon_occurrences.append(temp_dict['n'])
        
        answer = min(ballon_occurrences)        
        
        return answer 
    
solution = Solution()

print(solution.maxNumberOfBalloons(text = "nlaebolko"))

print(solution.maxNumberOfBalloons(text = "loonbalxballpoon"))

print(solution.maxNumberOfBalloons(text = "leetcode"))

print(solution.maxNumberOfBalloons(text = "lloo"))

File: my_project_typescript/easy_problems_typescript/from101to150/maximumNumberOfBalloons.ts
function maxNumberOfBalloons(text: string): number {

    // answer 
    let answer: number = 0

    // dictionary to add the total occurrences of each word in text
    let tempDict: any = {}

    let lenText: number = text.length

    // initialize the occurrences of each letter in the word text
    for (let word of "balon"){
        tempDict[word] = 0
    }

    // calculate how many time each word appears in text
    for (let i = 0; i < lenText; i++){
        tempDict[text[i]] = 0
    }

    for (let i = 0; i < lenText; i++){
        tempDict[text[i]] += 1
    }

    // the total amount of occurrences of each letter to form the word balloon
    let ballonOccurrences: Array<number> = []

    ballonOccurrences.push(tempDict['b'])
    ballonOccurrences.push(tempDict['a'])
    ballonOccurrences.push(Math.floor(tempDict['l']/2))
    ballonOccurrences.push(Math.floor(tempDict['o']/2))
    ballonOccurrences.push(tempDict['n'])


    answer = Math.min.apply(null, ballonOccurrences)

    return answer

};


console.log(maxNumberOfBalloons("nlaebolko"))

console.log(maxNumberOfBalloons("loonbalxballpoon"))

console.log(maxNumberOfBalloons("leetcode"))

console.log(maxNumberOfBalloons("lloo"))
